### PR TITLE
[Feature:Developer] Add yamllint linting to CI

### PR DESCRIPTION
**Pull request by williamjallen**
### What is the current behavior?
We currently have no YAML validation.  This can lead to improper YAML being introduced into our codebase as seen in https://github.com/Submitty/Submitty/pull/8788.  

### What is the new behavior?
This PR adds the [yamllint](https://yamllint.readthedocs.io/en/stable) tool to validate our YAML files automatically.  In practice, this will mostly help validate GitHub Actions workflow files, the dependabot configuration file, and the files which define the sample courses.

All of our existing failures have been addressed in this PR to make the linter pass successfully.

An associated documentation PR will be forthcoming soon.

### CODE REVIEW

**Comment by MasterOdin** [comment]
Hm, I'm not sure why the CI didn't run for this PR? Will need to investigate before we can merge this.

**Comment by MasterOdin** [comment]
An additional idea here is that we could use a json schema validator coupled with:
* https://json.schemastore.org/github-workflow.json
* https://json.schemastore.org/dependabot-2.0.json

which should also give us confidence that the schema is not only valid yaml syntax, but also is valid syntax for the tooling.

File: .github/workflows/submitty_ci.yml [hunk 1]
```diff
@@ -26,580 +26,632 @@ concurrency:
...
+        uses: actions/cache@v3
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-cache-node-modules-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+              ${{ runner.os }}-cache-node-modules-
+      - run: npm install
+      - run: npm run test
+      - name: Upload Coverage
+        uses: codecov/codecov-action@v1
```
Comment (MasterOdin) [inline .github/workflows/submitty_ci.yml:95]:
```suggestion
        uses: codecov/codecov-action@v3
```

File: .github/workflows/submitty_ci.yml [hunk 2]
```diff
@@ -26,580 +26,632 @@ concurrency:
...
+          npm install --production=false
+
+      - name: Run API tests
+        working-directory: ${{env.SUBMITTY_REPOSITORY}}/site
+        run: npm run test:api
+        env:
+          BASE_URL: http://localhost
+
+      - name: Run cypress e2e tests with pam auth
+        uses: cypress-io/github-action@v4
```
Comment (MasterOdin) [inline .github/workflows/submitty_ci.yml:529]:
```suggestion
        uses: cypress-io/github-action@v5
```

File: .github/workflows/submitty_ci.yml [hunk 3]
```diff
@@ -26,580 +26,632 @@ concurrency:
...
+          spec: cypress/e2e/autograding_status_2.spec.js
+          working-directory: ${{env.SUBMITTY_REPOSITORY}}/site
+          browser: chrome
+          env: autograding_shipper_off=True
+
+      - name: Switch to LDAP auth
+        run: sudo sed -ie "s/Database/Ldap/g" ${SUBMITTY_INSTALL_DIR}/config/authentication.json
+
+      - name: Run cypress e2e login tests with LDAP auth
+        uses: cypress-io/github-action@v4
```
Comment (MasterOdin) [inline .github/workflows/submitty_ci.yml:591]:
```suggestion
        uses: cypress-io/github-action@v5
```

**Review by IDzyre — APPROVED** [review]

**Review by williamjallen — COMMENTED** [review]
I approve these changes (I can't actually approve on GH since I created the original PR).

**Pull request merged**